### PR TITLE
Remove use of ENUMs in logic

### PR DESCRIPTION
**Pull request by larshp**
as per https://blogs.sap.com/2016/10/10/abap-news-release-7.51-enumerations/

> Normally, you are not interested in the contents of an enumerated object at all.

This PR removes use of ENUMS, as they are currently used for the purpose of their contents.

Note: I have not run the unit tests on a real system, `cl_oo_abap_doc_reader` does not exist on any system I have access to

### CODE REVIEW

**Comment by schneidermic0** [comment]
> This PR removes use of ENUMS, as they are currently used for the purpose of their contents.

Maybe, I don't get the reason for this PR. Why do you think they are used for the purpose of their contents? Except of the first, the don't even specify values:

```
  TYPES:
     BEGIN OF ENUM enum_formatting_option STRUCTURE formatting_option,
       no_formatting VALUE IS INITIAL,
       camel_case    VALUE 1,
     END OF ENUM enum_formatting_option STRUCTURE formatting_option.

   TYPES:
     BEGIN OF ENUM enum_type_info STRUCTURE type_info,
       string,
       numeric,
       boolean,
       date_time,
     END OF ENUM enum_type_info STRUCTURE type_info.

   TYPES:
     BEGIN OF ENUM enum_operation STRUCTURE operation,
       initial,
       write_element,
       open_structure,
       close_structure,
       open_table,
       close_table,
     END OF ENUM enum_operation STRUCTURE operation.
```

**Comment by larshp** [comment]
A: The unit test in https://github.com/SAP/abap-file-formats-tools/blob/cd44700a9924cc80c317067bf24c45bb44d21af4/src/zcl_aff_writer.clas.testclasses.abap#L236 relies on "initial" the first ENUM to be initial.

There is no assignment of the "initial" ENUM anywhere, so I'd guess it is never used.

However, INITIAL handing is strange in ENUMS, at best misleading, guess the output of this code,

```abap
TYPES:
  BEGIN OF ENUM enum_formatting_option STRUCTURE formatting_option,
    no_formatting VALUE IS INITIAL,
    camel_case    VALUE 1,
  END OF ENUM enum_formatting_option STRUCTURE formatting_option.
ASSERT formatting_option-no_formatting IS INITIAL.
WRITE formatting_option-no_formatting.
```

B: method [get_json_schema_type()](https://github.com/SAP/abap-file-formats-tools/blob/4490c25515d6ea5cfbd0c37af5c52a8503be401a/src/zcl_aff_writer_json_schema.clas.abap#L634) takes an ENUM as input in `json_type`, this enum is(depending on its value) `result = to_lower( type ).` which is the json output.

I.e. renaming the ENUMs will change the generated schema, ie. ENUMS are used for the purpose of their contents

**Comment by schneidermic0** [comment]
@wurzka and I had a look at this PR, again.

We think we have to distinguish between following aspects

A) code in abap-file-formats-tools shall not use enums so that it will be easier to using the transpiler for running unit tests in GitHub.

B) abap-file-formats-tools infrastructure does not support processing enums.

@larshp If we get you right, your goal is A only, isn't it? I.e., abap-file-formats-tools shall still be able to process enums.

We would agree to A if this enabled us to run more unit tests.

**Review by schneidermic0 — CHANGES_REQUESTED** [review]
> Note: I have not run the unit tests on a real system, cl_oo_abap_doc_reader does not exist on any system I have access to

We ran the tests in our system. There are a bunch of unit test errors.

**Comment by larshp** [comment]
yes, only "A"

**Comment by schneidermic0** [comment]
I suggest, we try to remove the unit test errors.

File: src/zcl_aff_writer.clas.testclasses.abap [hunk 1]
```diff
@@ -187,7 +187,7 @@ CLASS ltcl_type_writer IMPLEMENTATION.
   ENDMETHOD.
 
   METHOD get_type_info_string_like_enum.
-    cl_abap_unit_assert=>assert_equals( exp = zif_aff_writer=>type_info-string act = cut->get_json_type_from_description( get_element_description( VALUE zif_aff_writer=>enum_type_info( ) ) ) ).
+    cl_abap_unit_assert=>assert_equals( exp = zif_aff_writer=>type_info-string act = cut->get_json_type_from_description( get_element_description( '' ) ) ).
```
Comment (schneidermic0) [inline src/zcl_aff_writer.clas.testclasses.abap:190]: > yes, only "A"

In this case, we should revert this particular change. As far as I understand, this test shall verify, that the framework considers enums as `"type": "string"` in the JSON-schema.

Most probably, we have to define a local enum in the test and pass it to the method.

**Review by schneidermic0 — CHANGES_REQUESTED** [review]
See my comment + failing tests need to be fixed.

File: src/zcl_aff_writer.clas.testclasses.abap (hunk 1 above)
Comment (schneidermic0) [reply src/zcl_aff_writer.clas.testclasses.abap]: Added my suggestion with https://github.com/SAP/abap-file-formats-tools/pull/159/commits/045a1e88b1623dc10b70d54e08ca801fd6757a72

**Comment by schneidermic0** [comment]
abaplint errors are valid. However, I need the enum just for test reasons here. Can we add an exception somehow without changing the rule?

**Comment by larshp** [comment]
yup, skipped with https://github.com/SAP/abap-file-formats-tools/pull/159/commits/8bb5a573cfd7785ecc171cd7987eb300567b503d

**Review by schneidermic0 — APPROVED** [review]
Unit tests are green, now.

**Review by wurzka — APPROVED** [review]
LGTM

**Pull request merged**